src/user_prompts.ts

```diff
@@ -41,7 +41,77 @@ export function showUserPrompts(context: Context, workspaceContext: WorkspaceCon
 		promptToShowReleaseNotes(extensionVersion, versionLink).then(() =>
 			context.lastSeenVersion = extensionVersion,
 		);
+		return;
 	}
+
+	if (await showFlutter2019Q2SurveyNotificationIfAppropriate(context))
+		return; // Bail if we showed it, so we won't show any other notifications.
+
+	// (though, there are no other notifications right now...)
+}
+
+/// Shows Survey notification if appropriate. Returns whether a notification was shown
+/// (not whether it was clicked/opened).
+export function showFlutter2019Q2SurveyNotificationIfAppropriate(context: Context): boolean {
+	const now = Date.now();
+	// Mon May 13 2019 20:00:00 GMT+0100 (BST) = noon PDT on 13th May
+	const surveyStart = Date.UTC(2019, 4 /* Month is 0-based!! */, 13, 19, 0);
+	// Mon May 27 2019 08:00:00 GMT+0100 (BST) = midnight PDT between 26th/27th may.
+	const surveyEnd = Date.UTC(2019, 4 /* Month is 0-based!! */, 27, 7, 0);
+
+	if (now <= surveyStart || now >= surveyEnd)
+		return false;
+
+	const lastShown = context.flutterSurvey2019Q2NotificationLastShown;
+	const doNotShow = context.flutterSurvey2019Q2NotificationDoNotShow;
+
+	// Don't show this notification if user previously said not to.
+	if (doNotShow)
+		return false;
+
+	// Don't show this notification if we've shown it in the last 40 hours.
+	if (lastShown && now - lastShown < fortyHoursInMs)
+		return false;
+
+
```

Comment:
I think this extra space is problematic for `tslint`?

Reply:
Ha, good spot! One day I'll figure out why tslint issues seem to be unreliable showing in the editor... I'll fix this tomorrow when the other stuff is done, so it should all go green. Thanks!